src/transformers/tokenization_utils_base.py

```diff
@@ -1929,6 +1930,9 @@ def tojson(x, ensure_ascii=False, indent=None, separators=None, sort_keys=False)
             # We also expose some options like custom indents and separators
             return json.dumps(x, ensure_ascii=ensure_ascii, indent=indent, separators=separators, sort_keys=sort_keys)
 
+        def strftime_now(format):
+            return datetime.now().strftime(format)
+
```

Comment:
We shouldn't add functionalities like this into core tokenization code -- enabling certain things in a jinja template has nothing to do with tokenization and we risk bloating the module with many unrelated tools. Instead, I'd propose two things: 

* We move `strftime_now` to its own module we import from e.g. `jinja_template_utils` or `jinja_template_tools` or  [insert better name]. This would be a library where we can define all these different utilities. `raise_exception` should go there too. 
* We make the jinja template more extensible, such that we don't have to hard code this for everyone. It's been requested - but why expose this utility for all template users? It won't be a sustainable way to grow and manage this. Instead we could have e.g. a `jinja_globals` mapping which is used to add each named function e.g. 

```py
jinja_globals = {
    "raise_exception": raise_exception,
    "strftime_now": strftime_now
}
```

and then have model-specific defaults for their own tokenizers / templates, if we want.

The second suggestion I feel less strong about - very open to other suggestions on how to avoid adding all this logic inside the tokenizers

Reply:
Hmmn, moving it out of the tokenizer I agree with! However, I think I disagree with the model-specific `jinja_globals` mapping. The reason is that we'll have to hard-code everything users need as a Python function **somewhere**. We won't actually be able to let users add new functions in their model, because this is arbitrary Python code and would require `trust_remote_code=True`. 

Therefore, since we have to hardcode everything in the `transformers` codebase anyway, requiring specific extensions to be enabled in a model-specific `jinja_globals` adds confusion and extra model maintenance burden for no real gain - the only difference when an extension is enabled is that certain functions or keywords become usable inside Jinja. 

Therefore, I think we should just enable all of the extensions we support all the time. I don't think there is any performance penalty, so the only reason to disable an extension would be if users want to use a variable name that clashes with an extension, and we want to discourage that anyway.

This might change in future if we have a "dangerous" extension that poses a security risk, and so we want to limit it only to models that absolutely require it, to avoid a malicious template being able to call it. I have no plans to do anything like that right now, though, so I think that feature can wait until it's needed, if it ever is.